### DISCUSSION TITLE
two suggestions

### DISCUSSION THREAD

**Discussion by ac169** [microsoft/vcpkg#41946] [Ideas]
1. currently, only officially specified versions of packages can be installed. Some packages are not compatible with stable versions or commonly used versions. It is very complicated for developers to install and use specified versions. Is it possible to add the function of installing specified versions?

2. most of the official package original files are based on github, but some restricted areas cannot be directly accessed. Is it possible to increase the proxy download function and configuration parameters in an independent git environment?

**Comment by vipcxj** [comment]
Personally, I think it's a valuable suggestion. But it is unlikely to be realized for quite some time.
1. vcpkg uses git to store the port, you can specify different commits to lock all the dependency versions, but you can't just change the version of a dependency, that's the nature of git.
2. vcpkg although most of the dependencies are downloaded from github, but there are many other sources. That is to say, vcpkg's dependencies do not have a unified source, and where exactly to download from, is the port to decide, at present vcpkg is unlikely to unify the download source. But in theory it should be possible to create a proxy. Because no matter which download source, it will be cached in a unified location, but because the sources are so diverse, I'm not sure there are any legal issues.

**Comment by ac169** [comment]
> Personally, I think it's a valuable suggestion. But it is unlikely to be realized for quite some time.
> 
> 1. vcpkg uses git to store the port, you can specify different commits to lock all the dependency versions, but you can't just change the version of a dependency, that's the nature of git.
> 2. vcpkg although most of the dependencies are downloaded from github, but there are many other sources. That is to say, vcpkg's dependencies do not have a unified source, and where exactly to download from, is the port to decide, at present vcpkg is unlikely to unify the download source. But in theory it should be possible to create a proxy. Because no matter which download source, it will be cached in a unified location, but because the sources are so diverse, I'm not sure there are any legal issues.

The version is not a stable version, the difference between versions is too large, etc., so version management is a basic function of other common package management tools. As for git projects that rely on third-party projects, the version itself is also specified.

**Comment by dg0yt** [comment]
Versioning with vcpkg is expected to be done in manifest mode.
